cmd/telegraf/telegraf.go

```diff
@@ -122,7 +123,10 @@ func reloadLoop(
 		signal.Notify(signals, os.Interrupt, syscall.SIGHUP,
 			syscall.SIGTERM, syscall.SIGINT)
 		if *fWatchConfig != "" {
-			for _, fConfig := range fConfigs {
+			watchConfigs := make(sliceFlags, len(fConfigs))
+			copy(watchConfigs, fConfigs)
+			watchConfigs = append(watchConfigs, configDirFiles()...)
```

Comment:
`configDirFiles()...` doesn't look beautiful to me but I thought it would be a useless assignment and I wanted to confine the directory walking to a function as otherwise it would add a lot of lines here.

Reply:
It makes sense to put the directory walking into a separate function. I think the strangeness with seeing `configDirFiles()...` will go away if it returns a slice and an error. Then you will need to hold the slice in a variable for a line while you check the error.